components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -328,24 +330,22 @@ class QrFragment : Fragment() {
                 else -> false
             }
 
-            val displaySize = Point()
-            @Suppress("DEPRECATION")
-            // Deprecation of getDefaultDisplay() and getSize() will be handled in https://github.com/mozilla-mobile/android-components/issues/8518
-            activity?.windowManager?.defaultDisplay?.getSize(displaySize)
+            val displaySize = activity?.windowManager?.getDisplaySize()
+
             var rotatedPreviewWidth = width
             var rotatedPreviewHeight = height
-            var maxPreviewWidth = displaySize.x
-            var maxPreviewHeight = displaySize.y
+            var maxPreviewWidth = displaySize?.x
+            var maxPreviewHeight = displaySize?.y
 
             if (swappedDimensions) {
                 rotatedPreviewWidth = height
                 rotatedPreviewHeight = width
-                maxPreviewWidth = displaySize.y
-                maxPreviewHeight = displaySize.x
+                maxPreviewWidth = displaySize?.y
+                maxPreviewHeight = displaySize?.x
             }
 
-            maxPreviewWidth = min(maxPreviewWidth, MAX_PREVIEW_WIDTH)
-            maxPreviewHeight = min(maxPreviewHeight, MAX_PREVIEW_HEIGHT)
+            maxPreviewWidth = min(maxPreviewWidth ?: MAX_PREVIEW_WIDTH, MAX_PREVIEW_WIDTH)
+            maxPreviewHeight = min(maxPreviewHeight ?: MAX_PREVIEW_HEIGHT, MAX_PREVIEW_HEIGHT)
```

Comment:
We could to `activity?.windowManager?.getDisplaySize() ?: Point()`, that will remove all the null-able checks and match the previous implementation.